batch/batch_test.go

```diff
@@ -0,0 +1,86 @@
+package batch_test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/uw-labs/substrate"
+
+	"github.com/uw-labs/substrate-tools/batch"
+	"github.com/uw-labs/substrate-tools/message"
+	"github.com/uw-labs/substrate-tools/mock"
+)
+
+func TestBatchMessageSource(t *testing.T) {
+	t.Parallel()
+	tt := []struct {
+		Name         string
+		Messages     []substrate.Message
+		BatchSize    int
+		HandlerError error
```

Comment:
I find it hard to follow when mixing happy cases with error ones in test cases, due to the additional necessary ifs. I think it would be more easy to read if they would be separate.